### PR TITLE
live: markdown report support.

### PR DESCRIPTION
**Pull request by daavoo**
Per https://github.com/iterative/dvclive/pull/266

### CODE REVIEW

**Comment by daavoo** [comment]
Only added "API Reference" updates. Not sure (if) how to document the full "Live metrics" scenario with CML. 
Maybe a new `dvclive-with-cml` page?

File: content/docs/dvclive/get-started.md [hunk 1]
```diff
@@ -109,17 +109,20 @@ not.
...
 
+The `format` is determied by the `report` argument passed to
+[`Live()`](/doc/dvclive/api-reference/live#parameters).
+
 <admon type="info">
 
-If you don't update the step number, the HTML report won't be generated unless
-you call `Live.make_report()` directly.
+If you don't update the step number, the report won't be generated unless you
+call `Live.make_report()` directly.
```
Comment (dberenbaum) [inline content/docs/dvclive/get-started.md:125]: Do you prefer `report` or `metrics report`? I see `metrics report` everywhere but here. Can we make it consistent (no strong opinion on which one to use)?

File: content/docs/dvclive/api-reference/live/index.md [hunk 2]
```diff
@@ -38,11 +38,11 @@ other metadata.
 - `dir` - Location of the directory to store
   [outputs](/doc/dvclive/get-started#outputs).
 
-- `summary_path` - `{dir}.json`. Location of the
+- `summary_path` - `{Live.dir}.json`. Location of the
   [summary](/doc/dvclive/api-reference/live/log#description).
```
Comment (jorgeorpinel) [inline content/docs/dvclive/api-reference/live/index.md:42]: Hola. Why this change? Was just `dir` incorrect? It's listed as just `dir` above.

> Seems unrelated to https://github.com/iterative/dvclive/pull/266 too.

File: content/docs/dvclive/api-reference/live/index.md [hunk 3]
```diff
@@ -38,11 +38,11 @@ other metadata.
...
 
-- `html_path` - `{dir}/report.html`. Location of the
-  [html report](/doc/dvclive/api-reference/live/make_report#description).
+- `report_path` - `{Live.dir}/report.{format}`. Location of the
+  [metrics report](/doc/dvclive/api-reference/live/make_report#description).
```
Comment (jorgeorpinel) [inline content/docs/dvclive/api-reference/live/index.md:45]:
```suggestion
- `report_path` - `{Live.dir}/report.{format}`. Location of the
  [metrics report](/doc/dvclive/api-reference/live/make_report). The `{format}`
  can be `html` (HTML) or `md` (Markdown) depending on the value of the
  `report` parameter.
```

File: content/docs/dvclive/api-reference/live/index.md [hunk 4]
```diff
@@ -62,11 +62,14 @@ other metadata.
 
   </admon>
 
-- `report` - If `html`, DVCLive will call `Live.make_report()` on each step
-  update. _Default_: `html`.
+- `report` - If `auto`,`html` or `md`, DVCLive will call `Live.make_report()` on
+  each step update. _Default_: `auto`.
```
Comment (jorgeorpinel) [inline content/docs/dvclive/api-reference/live/index.md:66]: Please explain what `auto` does.

Comment (jorgeorpinel) [reply content/docs/dvclive/api-reference/live/index.md:66]: Nvmd I see it's down here...

File: content/docs/dvclive/api-reference/live/index.md [hunk 5]
```diff
@@ -62,11 +62,14 @@ other metadata.
...
 
-- `auto_open` - If `True`, on the first `Live.make_report()` call, DVCLive will
-  automatically open `html_path` in a browser. _Default_: `False`.
```
Comment (jorgeorpinel) [inline content/docs/dvclive/api-reference/live/index.md:69]: Wait what happened to `auto_open`?

File: content/docs/dvclive/api-reference/live/index.md (hunk 4 above)
Comment (jorgeorpinel) [inline content/docs/dvclive/api-reference/live/index.md:66]:
```suggestion
- `report` - If `auto`,`html`, or `md`, DVCLive will call `Live.make_report()` on
  each step update. The `auto` mode (default) will use `md` format if a `CI` env
  var is present, otherwise it will use `html`.
```

File: content/docs/dvclive/api-reference/live/index.md [hunk 6]
```diff
@@ -62,11 +62,14 @@ other metadata.
...
+  The `auto` mode will use `md` format if `CI` env var is present, otherwise, it
+  will use `html`.
+
+  If `report` is `None`, `Live.make_report()` won't be called on each step
+  update.
```
Comment (jorgeorpinel) [inline content/docs/dvclive/api-reference/live/index.md:72]:
```suggestion
  If `report` is `None`, `Live.make_report()` won't be called automatically.
```

**Review by jorgeorpinel — COMMENTED** [review]
Combined `report` param suggestion:

File: content/docs/dvclive/api-reference/live/make_report.md [hunk 7]
```diff
@@ -18,14 +18,17 @@ live.make_report()
 
 ## Description
 
-On each call, DVCLive will collect all the data logged in `{dir}`, generate an
-HTML report and save it in `{dir}/report.html`.
+On each call, DVCLive will collect all the data logged in `{Live.dir}`, generate
+a report and save it in `{Live.dir}/report.{format}`.
+
+The `format` is determied by the `report` argument passed to
+[`Live()`](/doc/dvclive/api-reference/live#parameters).
```
Comment (jorgeorpinel) [inline content/docs/dvclive/api-reference/live/make_report.md:25]:
```suggestion
The `format` The `format`
  can be HTML) or Markdown depending on the value of the `report
   argument passed to
   [`Live()`](/doc/dvclive/api-reference/live#parameters).
```

File: content/docs/dvclive/dvclive-with-dvc.md [hunk 8]
```diff
@@ -113,16 +113,16 @@ The [metrics summary](/doc/dvclive/api-reference/live/log#description) in
 The [metrics history](/doc/dvclive/api-reference/live/log#step-updates)
 `training_metrics/scalars` can be visualized with `dvc plots`.
 
-The [HTML report](/doc/dvclive/api-reference/live/make_report#description) in
+The [metrics report](/doc/dvclive/api-reference/live/make_report#description) in
```
Comment (jorgeorpinel) [inline content/docs/dvclive/dvclive-with-dvc.md:116]: 💅🏼 
```suggestion
The [metrics report](/doc/dvclive/api-reference/live/make_report) in
```

File: content/docs/dvclive/get-started.md [hunk 9]
```diff
@@ -109,17 +109,20 @@ not.
...
 
-If and when `step` is updated, DVCLive generates or updates an HTML report in
-`dvclive/report.html` which will contain all the logged data.
+If and when `step` is updated, DVCLive generates or updates a report in
+`dvclive/report.{format}` which will contain all the logged data.
 
 ![](/img/dvclive-html.gif)
 
+The `format` is determied by the `report` argument passed to
+[`Live()`](/doc/dvclive/api-reference/live#parameters).
```
Comment (jorgeorpinel) [inline content/docs/dvclive/get-started.md:120]:
```suggestion
The `format` can be HTML) or Markdown depending on the value of the `report`
argument passed to
[`Live()`](/doc/dvclive/api-reference/live#parameters).
```

File: content/docs/dvclive/api-reference/live/index.md (hunk 2 above)
Comment (daavoo) [reply content/docs/dvclive/api-reference/live/index.md:42]: > Seems unrelated to https://github.com/iterative/dvclive/pull/266 too.

It has nothing to do, just something I noticed during this P.R. Could be extracted to other

File: content/docs/dvclive/api-reference/live/index.md (hunk 5 above)
Comment (daavoo) [reply content/docs/dvclive/api-reference/live/index.md:69]: Docs were outdated. I just noticed this was still here during this P.R. 
It was removed a few versions ago, discussion here https://github.com/iterative/dvclive/issues/241

**Review by dberenbaum — APPROVED** [review]

**Comment by daavoo** [comment]
@jorgeorpinel merging the current state. If there are any suggestions left, please comment and I will address them in a follow-up

**Review by casperdcl — COMMENTED** [review]
late review

**Comment by jorgeorpinel** [comment]
Did you have a chance to follow-up on @casperdcl's comments @daavoo ? Maybe you can include any updates that still apply to #3923.

**Comment by jorgeorpinel** [comment]
Thank you!
